django/apps/registry.py

```diff
@@ -344,6 +346,79 @@ def clear_cache(self):
             for model in self.get_models(include_auto_created=True):
                 model._meta._expire_cache()
 
+    @staticmethod
+    def _model_key(model_or_name):
+        """
+        Returns an (app_label, model_name) tuple from a model or string.
+        """
+        if isinstance(model_or_name, six.string_types):
+            app_label, model_name = model_or_name.split(".")
+        else:
+            # It's actually a model class.
+            app_label = model_or_name._meta.app_label
+            model_name = model_or_name._meta.object_name
+        return app_label, model_name
+
+    def lazy_model_operation(self, function, *models_or_names, **kwargs):
+        """
+        Take a function and a number of models or model names, and when all
+        the corresponding models have been loaded, call the function with
+        the model classes and any optional as its arguments.
+
+        Each element of models_or names may be
+            * an actual model class
+            * a string of the form "app_label.ModelName"
+            * a tuple of strings of the form (app_label, model_name)
+
+        The function passed to this method must accept n positional arguments,
+        where n=len(models_or_names), plus whatever keyword arguments you pass
+        in kwargs.
+        """
+
+        if not self._class_prepared_handler_registered:
```

Comment:
You're right, we don't. The use of the ``class_prepared`` is an artefact from this code's past as a standalone class in a third-party app, happy to cut it out.